core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -44,9 +62,34 @@ public function createResponseContext(): ResponseContext
 
     public function createContaoWebpageResponseContext(PageModel $pageModel): ContaoWebpageResponseContext
     {
-        $context = new ContaoWebpageResponseContext($pageModel);
+        $context = new ContaoWebpageResponseContext(new JsonLdManager($this->eventDispatcher), $pageModel);
         $this->responseContextAccessor->setResponseContext($context);
 
         return $context;
     }
+
+    public function onJsonLd(JsonLdEvent $event): void
```

Comment:
I don't know why you ask me this at the exact spot where I wrote that I find this "a beautiful design decision"?
So again, no I don't think it should go to its separate listener because again, I find this a beautiful design decision to have the things together here.